pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
Ok, then i'm fine with this being very closely coupled here, and not end
user configurable.

On Tue, Feb 2, 2016 at 1:29 PM, Avesh Agarwal <notifications@github.com>
wrote:

> In pkg/cmd/server/kubernetes/node_config.go
> <https://github.com/openshift/origin/pull/6538#discussion_r51611737>:
>
> > @@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
> >  	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
> >  	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
> >
> > +	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
> > +		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
>
> As I understand, SerializeImagePulls was introduced due to issue with 1.8
> with parallel pulls. If parallel-image-pulls works fine with any/future
> version of docker, then why would someone want to use serialpulls, I am not
> sure, doesn't make much sense though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6538/files#r51611737>.
>